### PR TITLE
refactor code structure of pruning algorithms

### PR DESCRIPTION
**Pull request by QuanluZhang**

### CODE REVIEW

File: src/sdk/pynni/nni/compression/torch/activation_rank_filter_pruners.py [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+
+logger = logging.getLogger('torch activation rank filter pruners')
+
+class ActivationRankFilterPruner(Pruner):
+    """
+    A structured pruning base class that prunes the filters with the smallest
+    importance criterion in convolution layers to achieve a preset level of network sparsity.
+    Hengyuan Hu, Rui Peng, Yu-Wing Tai and Chi-Keung Tang,
+    "Network Trimming: A Data-Driven Neuron Pruning Approach towards Efficient Deep Architectures", ICLR 2016.
+    https://arxiv.org/abs/1607.03250
```
Comment (chicm-ms) [inline src/sdk/pynni/nni/compression/torch/activation_rank_filter_pruners.py]: Can we remove the paper description and reference as this base class does not implement the algos ?

Comment (QuanluZhang) [reply src/sdk/pynni/nni/compression/torch/activation_rank_filter_pruners.py]: good point, thanks.

File: src/sdk/pynni/nni/compression/torch/activation_rank_filter_pruners.py [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+
+            def _hook(module_, input_, output, name=layer.name):
+                if len(self.collected_activation[name]) < self.statistics_batch_num:
+                    self.collected_activation[name].append(self.activation(output.detach().cpu()))
+
+            layer.module.register_forward_hook(_hook)
+        return self.bound_model
+
+    def _get_mask(self, base_mask, activations, num_prune):
+        return {'weight': None, 'bias': None}
```
Comment (chicm-ms) [inline src/sdk/pynni/nni/compression/torch/activation_rank_filter_pruners.py]: Can we raise NotImplementedError here instead?

Comment (QuanluZhang) [reply src/sdk/pynni/nni/compression/torch/activation_rank_filter_pruners.py]: good point!

Comment (QuanluZhang) [reply src/sdk/pynni/nni/compression/torch/activation_rank_filter_pruners.py]: which func name do you think is better? `_get_mask` or `get_mask`?

Comment (chicm-ms) [reply src/sdk/pynni/nni/compression/torch/activation_rank_filter_pruners.py]: `get_mask` is consistent with calc_mask,   they are both from base class.

File: src/sdk/pynni/nni/compression/torch/weight_rank_filter_pruners.py [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+
+import logging
+import torch
+from .compressor import Pruner
+
+__all__ = ['L1FilterPruner', 'L2FilterPruner', 'FPGMPruner']
+
+logger = logging.getLogger('torch weight rank filter pruners')
+
+class WeightRankFilterPruner(Pruner):
```
Comment (chicm-ms) [inline src/sdk/pynni/nni/compression/torch/weight_rank_filter_pruners.py]: `WeightRankFilterPruner` is a little bit complex,  maybe `FilterPruner`?

File: src/sdk/pynni/nni/compression/torch/activation_rank_filter_pruners.py (hunk 2 above)
Comment (QuanluZhang) [reply src/sdk/pynni/nni/compression/torch/activation_rank_filter_pruners.py]: updated, please help review again

File: src/sdk/pynni/nni/compression/torch/weight_rank_filter_pruners.py (hunk 3 above)
Comment (QuanluZhang) [reply src/sdk/pynni/nni/compression/torch/weight_rank_filter_pruners.py]: I think `WeightRank` is the emphasis.

Comment (chicm-ms) [reply src/sdk/pynni/nni/compression/torch/weight_rank_filter_pruners.py]: Ok, that's fine for me.

**Pull request merged**